core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -99,12 +115,20 @@ public int read() throws IOException {
     if (mPosition == mLength) { // at end of file
       return -1;
     }
-    updateStream();
-    int result = mBlockInStream.read();
-    if (result != -1) {
-      mPosition++;
+
+    resetRetry();
```

Comment:
The `resetRetry()` is only for resetting the state of retry count and whether to retry worker. The list of lost worker is retained for the life of the stream. It will not retry on worker that is added to lost workers list in previous reads. It will reset the counter if previous read is successful so it will keep progressing if the error is intermittent as long as the read can succeed after several retries.

Reply:
oh I see, that makes more sense